service/history/workflow/task_generator.go

```diff
@@ -161,6 +165,7 @@ func (r *TaskGeneratorImpl) GenerateWorkflowStartTasks(
 
 func (r *TaskGeneratorImpl) GenerateWorkflowCloseTasks(
 	now time.Time,
+	closeEvent *historypb.HistoryEvent,
```

Comment:
You can call mutableState.GetWorkflowCloseTime() and pass in the close time

Reply:
Yeah that's my first attempt on the implementation. However `GetWorkflowCloseTime` requires a context parameter, which means I still need to modify the interface to take in a context value, but currently none of the methods in task generator has context parameter. 

Only task refresh command needs to get the close event from DB, which I think is ok. In all other places, close event is already available when calling this method, so pass in the close event won't have any extra cost.

Reply:
I can change it back to use `GetWorkflowCloseTime` or use executionInfo.CloseTime directly when we stop supporting getting close time from close event. I can add a TODO comment here.

Reply:
Actually I already added a TODO for that on L52